TypeCobol/Compiler/CompilationUnit.cs

```diff
@@ -286,7 +286,17 @@ public void RefreshCodeAnalysisDocumentSnapshot()
 
                     List<Diagnostic> diagnostics = new List<Diagnostic>();
                     Dictionary<string, object> results = new Dictionary<string, object>();
-                    var analyzers = _analyzerProvider?.CreateQualityAnalyzers(CompilerOptions);
+                    IQualityAnalyzer[] analyzers = null;
+                    try
+                    {
+                        analyzers = _analyzerProvider?.CreateQualityAnalyzers(CompilerOptions);
+                    }
+                    catch (Exception exception)
+                    {
+                        var diagnostic = new Diagnostic(MessageCode.AnalyzerFailure, Diagnostic.Position.Default, $"Error while creating quality analyzers with provider : {_analyzerProvider}", exception.Message, exception);
```

Comment:
`_analyzerProvider` may be null here. I think you should create a local function `TryCreateQualityAnalyzers` and start with a code like `if (_analyzerProvider != null && TryCreateQualityAnalyzers(out var analyzers)` and then continue with the analysis itself.